kolibri/plugins/user/assets/src/views/SignInPage/index.vue

```diff
@@ -347,7 +349,38 @@
           this.setSuggestionTerm(newVal);
         }
       },
+      // Watch these computed properties and focus the fields
+      // that need to be focused for cleaner transitions
+      showPasswordForm(b) {
+        if (b) {
+          this.$nextTick(() => this.$refs.password.focus());
+        }
+      },
+      showUsernameForm(b) {
+        if (b) {
+          this.$nextTick(() => this.$refs.username.focus());
+        }
+      },
+    },
+    // Clear the username when entering the route.
+    // username may be held over if you select a user from UsersList
+    // then change to a facility that doesn't use UsersList
+    //
+    // TODO: If we want to clear the username whenever we switch facilities,
+    // then we can remove the `beforeRouteEnter` and use the commented out
+    // `beforeRouteLeave` below
+    beforeRouteEnter(to, from, next) {
```

Comment:
Pretty sure this clears it when we switch facilities because we clear it onRouteEnter - will clarify this section's comments later.